### DISCUSSION TITLE
YARP distro packages list

### DISCUSSION THREAD

**Discussion by ipa-rar** [robotology/community#497] [Q&A] [answered]
Does YARP has a distribution.yaml or source.yaml file like in ROS that contains the list of all the released packages. I found [this link](https://icub-tech-iit.github.io/documentation/sw_versioning_table/2021.02.feat-01/). Is this the final list of released packages.

**Comment by pattacini** [comment]
Hi @ipa-rar,

YARP is a component of the [distros][1] we've been releasing through the workflow put in place through [`robotology/robotology-superbuild`][2], as it comes out from the list you pointed at. In detail, you can find our distros' distribution lists at https://github.com/robotology/robotology-superbuild/tree/master/releases.

The YARP framework itself is made of several libraries and components (`os`, `sig`, device drivers etc). I don't think there's a sort of YAML-based equivalent list for YARP though. @drdanz correct me if I'm wrong.

[1]: https://icub-tech-iit.github.io/documentation/sw_versioning_table/#table
[2]: https://github.com/robotology/robotology-superbuild

**Comment by traversaro** [comment, marked as answer]
Hi @ipa-rar, to complement what @pattacini wrote, what are you interested in specifically? 

To clarify, YARP is a classical C++ package composed by a a few libraries and tools all maintained at https://github.com/robotology/yarp. On the other hand, ROS is a term that is used to refer to several mutually related aspects and projects. Among this aspects and projects, one project is the ROS distributions, that are a secondary distributions of software mainly based on Ubuntu repositories, but also other systems.  I used the term "secondary distribution" because as opposed to traditional Linux distribution that package and mantain all the software stack down to the operating system kernel, the ROS Distribution only maintain and package a set of ROS and Robotics related software, on the top of a "primary" distribution that provides the rest of the software. Historically the official primary distribution of ROS distributions has been Ubuntu (see https://www.ros.org/reps/rep-0003.html#noetic-ninjemys-may-2020-may-2025), but other primary distributions has been targeted with various level of support or official endorsement. In this sense, the ROS distributions and the YARP library are not really comparable projects at all. YARP is just a software package that could be packaged by any distribution, either primary or secondary.

As @pattacini mentioned, to easily support running software related to the iCub robot, we (iCub mantainers) are maintaining a small (much smaller then ROS, to clarify) secondary distribution at https://github.com/robotology/robotology-superbuild , and the latest releases for the packages contained there can be found in https://github.com/robotology/robotology-superbuild/blob/master/releases/latest.releases.yaml (in the same .yaml format supported by the [vcstool](https://github.com/dirk-thomas/vcstool) used in the ROS world), while we also do "distro" releases every three months that can be found in https://github.com/robotology/robotology-superbuild/tree/master/releases . However, this are not in any way the "YARP packages", as it includes the YARP C++ package and many other software packages.
